source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_hw.c

```diff
@@ -47,17 +47,16 @@ void setup_isr( xemacpsif_s * xemacpsif )
     /*
      * Setup callbacks
      */
-    XEmacPs_SetHandler( &xemacpsif->emacps, XEMACPS_HANDLER_DMASEND,
-                        ( void * ) emacps_send_handler,
-                        ( void * ) xemacpsif );
```

Comment:
I missed this change: why would you remove the function calls and replace them with direct assignments?
Was there a problem with the functions? Or did it introduce a dependency on sources?

Reply:
Hello @ChristosZosi,
I agree with Hein here. I am also not sure why was this removed?

Reply:
In the commit message (9829705528a353d340b68fe901074105a3d3c257) I did add an explanation why the calls to the `XEmacPs_SetHandler` is problematic. In short, the compiler would output the pedantic warning `ISO C forbids conversion of object pointer to function pointer type`, when compiled with `-Wpedantic`. 

The reason for it has to do with the function prototype of  `XEmacPs_SetHandler`
```
LONG XEmacPs_SetHandler(XEmacPs *InstancePtr, u32 HandlerType,
			void *FuncPointer, void *CallBackRef);
``` 

Here you can see that the third parameter of the function is a `void`-pointer. However we pass to the function a function pointer. And here lies the issue, that function pointers (text) are not the same as void pointers (data). That is the reason for the compiler warning.

Here is also a nice discussion on stack overflow, which addresses this exact issue https://stackoverflow.com/questions/36645660/why-cant-i-cast-a-function-pointer-to-void 

I hope this explains it 😄.

Reply:
Thank you for the URL and for the insights.

Actually I had the same problem when using `setsockopt()` which has a parameter `void *optval`. It was not allowed to pass a function pointer.

As a solution, I introduced a struct that holds a function pointer called [F_TCP_UDP_Handler_t](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/afcedead21c747cef64f07c7fedd50df75bcbd10/source/include/FreeRTOS_Sockets.h#L432).